app/services/custom_fields/hierarchy/hierarchical_item_service.rb

```diff
@@ -97,11 +97,24 @@ def move_item(item:, new_parent:)
 
       # Reorder the item along its siblings.
       # @param item [CustomField::Hierarchy::Item] the parent of the node
-      # @param new_sort_order [Integer] the new parent of the node
+      # @param new_sort_order [Integer] the new position of the node
       # @return [Success(CustomField::Hierarchy::Item)]
       def reorder_item(item:, new_sort_order:)
-        old_item = item.siblings.where(sort_order: new_sort_order).first
-        Success(old_item.prepend_sibling(item))
+        return Success(item) if item.siblings.empty?
+
+        new_sort_order = [0, new_sort_order].max
+
+        return Success(item) if item.sort_order == new_sort_order
+
+        target_item = item.siblings.find_by(sort_order: new_sort_order)
+        if target_item.present?
+          target_item.prepend_sibling(item)
+        else
+          target_item = item.siblings.last
+          target_item.append_sibling(item)
+        end
+
+        Success(item.reload)
```

Comment:
🟡 I would guess this database query is not necessary as the return values of this method is not read.

Also, what happens, if the ordering fails for some reason?

Reply:
> this database query is not necessary as the return values of this method is not read

I am not sure what to do with this one. It is a bit more defensive to keep the reload if in the future we use this action for something else that depends on the response. But on the other hand it's an extra unnecessary query.

We could replace for an empty `Success`, similar to the `delete_branch` method.

...

> what happens, if the ordering fails for some reason?

I did not think of a path where this operation fails :grimacing:. The worse that can happen is that no reordering happens, but the operation itself is still successful, I think.

Reply:
Okay, just for documenting here, I went with removing the object from the success response.